### PR TITLE
Add app-text/dblatex and fix app-text/asciidoc deps

### PR DESCRIPTION
**Pull request by marcecj**
This small patch series attempts to fix some dependency issues with app-text/asciidoc, as detailed in [bug \#366763](https://bugs.gentoo.org/show_bug.cgi?id=366763).  The first  patch adds an ebuild for app-text/dblatex, which is required by the a2x tool for generating PDFs.  The second patch then revbumps app-text/asciidoc to add a dependency on dblatex and on www-client/w3m or www-client/lynx, one of which is needed for the text output of a2x to work (i.e., `a2x -f text`).

Now, I expect this PR to be a bit more complicated, since a new ebuild is involved and somebody needs to maintain it, after all.  However, the ebuild is relatively uncomplicated, and the patches (if they're even really necessary?) don't appear to need much updating -- neither has changed since 2012.  In any case, my intent here is to get the ball rolling :-) .

FWIW, I've been using a variant of the dblatex ebuild since 2012 (dblatex-0.3.2) and have upgraded to the ebuild from this PR today, with no ill effects.

### CODE REVIEW

File: app-text/dblatex/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<longdescription lang="en">
+	</longdescription>
+</pkgmetadata>
+
```
Comment (mgorny) [inline app-text/dblatex/metadata.xml]: Extraneous empty line. Plus you need to add someone as maintainer here. Likely you ;-).

Comment (mgorny) [inline app-text/dblatex/metadata.xml]: Plus `<herd>proxy-maintainers</herd>`, unless somoene else volunteers to maintain it.

File: app-text/dblatex/files/dblatex-0.3.7-path-logging.patch [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# from the wagnerflo overlay, see
+# https://bugs.gentoo.org/show_bug.cgi?id=129368
```
Comment (mgorny) [inline app-text/dblatex/files/dblatex-0.3.7-path-logging.patch]: Is this going upstream?

File: app-text/dblatex/dblatex-0.3.7.ebuild [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+inherit distutils-r1
+
+DESCRIPTION="Transform DocBook using TeX macros"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.bz2"
+HOMEPAGE="http://dblatex.sourceforge.net/"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~x86 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~x86-freebsd ~amd64-linux ~arm-linux ~x86-linux ~ppc-macos ~x86-macos ~x86-solaris"
```
Comment (mgorny) [inline app-text/dblatex/dblatex-0.3.7.ebuild]: Quite a set of machines you've got there. Unless you don't. In that case, put only the arches you actually tested.

File: app-text/dblatex/dblatex-0.3.7.ebuild [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	dev-texlive/texlive-pictures"
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"$FILESDIR/$P-install_layout.patch"
+	"$FILESDIR/$P-path-logging.patch"
+)
+
+src_install() {
+	distutils-r1_src_install || die "distutils install failed"
```
Comment (mgorny) [inline app-text/dblatex/dblatex-0.3.7.ebuild]: ||die is implicit in the eclass helpers, you don't need to do that.

File: app-text/dblatex/Manifest [hunk 5]
```diff
@@ -0,0 +1,5 @@
+AUX dblatex-0.3.7-install_layout.patch 502 SHA256 53ad2207b9f1181f16e7fe8bae919d5a74f4d22ba15945c61322a0fda06472f0 SHA512 e8ad310d83425f18f1711452a2c23d7159807a323faf3fc698c554548fac5dd1d1ac634ab13d7a7cd0f87fc4258ec0b833370424ec687df9cef290a802b41174 WHIRLPOOL 15fd2d116272adecb271533111a40469274005c3f9b72b05c95ed42cf739c95e62d8504509ee0836df9fc855ac9a6e2c311b648c478251a3159caff45776c6b7
```
Comment (mgorny) [inline app-text/dblatex/Manifest]: This is not a valid thin Manifest. Which indicates you copied it from some overlay without regenerating.

**Comment by mgorny** [comment]
Convenient diff:

```
--- asciidoc-8.6.9-r1.ebuild	2015-08-09 10:19:37.048932801 +0200
+++ asciidoc-8.6.9-r2.ebuild	2015-10-26 23:49:51.315230890 +0100
@@ -11,7 +11,7 @@ inherit python-single-r1
 DESCRIPTION="A text document format for writing short documents, articles, books and UNIX man pages"
 HOMEPAGE="http://asciidoc.org/"
 SRC_URI="mirror://sourceforge/project/${PN}/${PN}/${PV}/${P}.tar.gz"
-KEYWORDS="alpha amd64 arm ~arm64 hppa ia64 ~mips ppc ppc64 ~s390 ~sh sparc x86 ~amd64-fbsd ~x86-fbsd ~x86-freebsd ~amd64-linux ~arm-linux ~x86-linux ~ppc-macos ~x86-macos ~x86-solaris"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~x86-freebsd ~amd64-linux ~arm-linux ~x86-linux ~ppc-macos ~x86-macos ~x86-solaris"
 
 LICENSE="GPL-2"
 SLOT="0"
@@ -23,6 +23,8 @@ RDEPEND=">=app-text/docbook-xsl-styleshe
 		dev-libs/libxslt
 		graphviz? ( media-gfx/graphviz )
 		app-text/docbook-xml-dtd:4.5
+		app-text/dblatex
+		|| ( www-client/w3m www-client/lynx )
 		highlight? ( || ( dev-python/pygments[${PYTHON_USEDEP}] dev-util/source-highlight ) )
 		${PYTHON_DEPS}
 "
```

**Comment by mgorny** [comment]
Please also apply the changes to -9999.

**Comment by mgorny** [comment]
@djc, asciidoc's your domain. Not sure if you want to take the extra dep as well.

File: app-text/dblatex/files/dblatex-0.3.7-path-logging.patch (hunk 2 above)
Comment (marcecj) [inline app-text/dblatex/files/dblatex-0.3.7-path-logging.patch]: I can try to get it upstream, sure (after I look at it more closely, that is).

File: app-text/dblatex/dblatex-0.3.7.ebuild (hunk 3 above)
Comment (marcecj) [inline app-text/dblatex/dblatex-0.3.7.ebuild]: I'll change that.  I mean, I did that originally (well, ~amd64 and ~x86), but then repoman spat out fatal errors like

      dependency.bad [fatal]        44
       app-text/asciidoc/asciidoc-8.6.9-r2.ebuild: RDEPEND: ~alpha(default/linux/alpha/13.0)
    ['app-text/dblatex']

so I copied asciidoc's keywords.  I decided to check the devmanual and first it says "An ebuild must not depend upon any package that is of a lower keyword level than itself." (which I guess is where the repoman error comes from), but then further down under "Keywording New Packages" it says "Just file bugs for these architectures when a package is going to be a dependency of a package already keyworded.".  I take it this this what you are alluding to?

File: app-text/dblatex/dblatex-0.3.7.ebuild (hunk 4 above)
Comment (marcecj) [inline app-text/dblatex/dblatex-0.3.7.ebuild]: OK, I'll fix that.

File: app-text/dblatex/Manifest (hunk 5 above)
Comment (marcecj) [inline app-text/dblatex/Manifest]: Ugh, sorry about that, I'll fix it.

**Comment by marcecj** [comment]
OK, I added further commits following the comments from @mgorny.  I can squash them later if requested (which I think is a good idea, but in principal I'm also fine with not squashing them).

File: app-text/dblatex/metadata.xml (hunk 1 above)
Comment (marcecj) [inline app-text/dblatex/metadata.xml]: I would rather wait first to see if somebody else wants to maintain it.

In principal I'm not against the idea of being a proxy-maintainer, but I'd have to look into what's involved first.  I don't want to promise something I can't keep.

File: app-text/dblatex/files/dblatex-0.3.7-path-logging.patch (hunk 2 above)
Comment (mgorny) [inline app-text/dblatex/files/dblatex-0.3.7-path-logging.patch]: We always try to get patches upstream, if they are suitable. If they're not, we try to improve them ;-)

Comment (marcecj) [inline app-text/dblatex/files/dblatex-0.3.7-path-logging.patch]: Alright, I opened https://sourceforge.net/p/dblatex/patches/9/.  Feel free to comment if you have anything to add.

**Comment by djc** [comment]
I don't really want to be maintaining asciidoc that much.@marcecj, would you like to be proxy maintainer for both asciidoc and dblatex?

Also, I'm a bit concerned about the extra dependencies. Is there some sensible way that we can put the dependencies (probably including this entire `a2x` tool?) behind a USE flag?

**Comment by idella** [comment]
I know this is wanted as a required dep to support another recently added package. It's djc who introduced the notion of proxy maintaining which can be done directly under djc or the proxy-maintainers project. Either way a decision needs making and the metadata.xml completing.

**Comment by marcecj** [comment]
I'm sorry for taking my time on this, but I don't have a definitive answer yet.  I'm not overly excited about the idea, but at the same time I'd like this fixed.  I would argue that I lack the prerequisite of being enthusiastic about the work involved with proxy-maintaining.  On the other hand, neither asciidoc nor dblatex are particularly fast-paced packages, so I can imagine that they should be easily manageable.

But another reason I was waiting was because I thought that maybe a Gentoo dev would want to pick up these two packages.  If that is, as it seems, not the case, then I would be a bit more inclined to accept the offer to proxy-maintain them.

In the meantime I will continue to think about it.

But while I'm writing this: @djc, I skimmed over the `Makefile.in`, and both asciidoc and a2x are always lumped together, and the configure script doesn't provide any relevant options, so it looks like the build system does not directly support leaving out a2x.  However, I suppose that it might be possible to delete a2x before the merge step.  I haven't tried it, but would that be considered OK, or is it too hacky?

**Comment by idella** [comment]
It's unlikely a developer is looking to swoop on these. If you select to be a proxy maintainer, you get the support from members if the project that comes with it, predominantly in the irc channel. The "membership" has had a recent surge. Still, proxied users maintainer according to their available time and interest drives them.

**Comment by marcecj** [comment]
Huh.  Looks like I can drop the dblatex ebuild: https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=337e2f7b91b016c1f1181e2125c0e74da7497ba4.  @zpuskas is its proxy maintainer.

In light of this, I think I'll take up the offer of proxy maintainer for asciidoc (unless @zpuskas wants it really badly ;-) ).

**Comment by djc** [comment]
I think it would be great if you take on asciidox proxy maintenance!

**Comment by marcecj** [comment]
I just want to be sure I'm not misunderstood: I will accept the role of proxy maintainer if you think I can do it. I may be a user of asciidoc, but only in one project. I do not have any knowledge on docbook, and only minimal knowledge of XML. If the latter (docbook) is necessary, then I am most likely not well suited to the role.

**Comment by zpuskas** [comment]
@marcecj Yeah, feel free to take on asciidoc, I don't mind. Also have other packages on my plate to pick up :)

**Comment by idella** [comment]
So marcecj can you re-do all your commits to reduce down to just asiidoc and let's start afresh, or close this and make a new pull req

**Comment by marcecj** [comment]
I was going to open a new PR after @djc responds (I don't expect a change in his opinion, but I wanted to be sure), but I can do it now.

**Comment by djc** [comment]
No change in my opinion, please go ahead. :)

**Comment by marcecj** [comment]
Alright, I've opened PR #344, which obsoletes this PR.

**Pull request closed without merging**